docs/datastore/react.md

```diff
@@ -0,0 +1,179 @@
+---
+id: react
+title: React - using Datastore Hooks
+sidebar_label: React
+---
+
+Offix Datastore provides hooks for CRUD operations and subscription.
+
+For a quick start see [sample react app](https://github.com/aerogear/offix/tree/master/examples/react-datastore).
+
+## useSave
+
+The `useSave` hook provides a lazy `save` function,
```

Comment:
Question - how useSave will work with saveById?

Reply:
I think we will need new hooks. WDYT?

Reply:
definetly need new issue. We cah use something like `upsert:true` 

And the way things are updated now is really strange you can pass array (does it mean you want to remove all objects). I think our hooks should be by id only for the moment, but need to check how we specify id (passing entire object vs id + effect)

Reply:
Where do we pass array? Our api doesn't accept arrays anywhere(that I know of). I think It's okay for us to restrict update and delete to by Id. Also, the way we specify id doesn't matter much IMO. Passing entire object could be better if you are look at how users send updates to API. 
If you support passing entire object and restricting update and delete to by id, I can push this in new PR. Also, having and `upsert` option for `useUpdate` hook seems like the best approach right now. Since, you approve, I can push that too.

Reply:
Let's create issue to chat about it, but feel free to get upset as quick win

Reply:
Let's create issue to chat about it, but feel free to get upset as quick win